drivers/platform/maxim/maxim_gpio.c

```diff
@@ -0,0 +1,439 @@
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include <errno.h>
+#include <stdlib.h>
+#include "no-os/gpio.h"
+#include "no-os/irq.h"
+#include "no-os/util.h"
+#include "gpio.h"
+#include "gpio_regs.h"
+#include "gpio_extra.h"
+#include "max32660.h"
+
+#define N_INT	14
+
+static struct callback_desc *gpio_callback[N_INT];
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+void GPIO0_IRQHandler()
+{
+	uint32_t pin = 0;
+	mxc_gpio_regs_t *gpio_regs = MXC_GPIO_GET_GPIO(0);
+	uint32_t stat_reg = gpio_regs->int_stat;
+
+	/** Clear interrupt flags */
+	gpio_regs->int_clr = stat_reg;
+	while(stat_reg) {
+		if(stat_reg & 1) {
+			if(!gpio_callback[pin])
+				return;
+			void *ctx = gpio_callback[pin]->ctx;
+			gpio_callback[pin]->callback(ctx, pin, NULL);
+		}
+		pin++;
+		stat_reg >>= 1;
+	}
+}
+
+/**
+ * @brief Obtain the GPIO decriptor.
+ * @param desc - The GPIO descriptor.
+ * @param param - GPIO initialization parameters
+ * @return 0 in case of success, errno error codes otherwise.
+ */
+int32_t max_gpio_get(struct gpio_desc **desc,
+		     const struct gpio_init_param *param)
+{
+	if(!param || param->number >= N_PINS)
+		return -EINVAL;
+
+	struct gpio_desc *descriptor = calloc(1, sizeof(*descriptor));
+	if(!descriptor)
+		return -ENOMEM;
+
+	descriptor->number = param->number;
+	descriptor->platform_ops = param->platform_ops;
+	descriptor->extra = param->extra;
```

Comment:
To use this properly the user will have to know how this driver works as well as the maxim DFP driver and know the name of the `gpio_cfg_t` object the they need to place in the initialization structure. Is there a way to infer it based on the pin number or maybe a port ID? Those things should be more straight forward to use.